src/Tests/main.test.ts

```diff
@@ -0,0 +1,59 @@
+import * as core from "@actions/core";
+import main from "../main";
+import { AuthorizerFactory } from 'azure-actions-webclient/AuthorizerFactory';
+import { AzureAppServiceUtility } from 'azure-actions-appservice-rest/Utilities/AzureAppServiceUtility';
+import { ContainerDeploymentUtility } from 'azure-actions-appservice-rest/Utilities/ContainerDeploymentUtility';
+import { KuduServiceUtility } from 'azure-actions-appservice-rest/Utilities/KuduServiceUtility';
+import { AzureAppService } from 'azure-actions-appservice-rest/Arm/azure-app-service';
+import { TaskParameters } from "../taskparameters";
+
+jest.mock('@actions/core');
+jest.mock('../Taskparameters');
+jest.mock('azure-actions-webclient/AuthorizerFactory');
+jest.mock('azure-actions-appservice-rest/Arm/azure-app-service');
+jest.mock('azure-actions-appservice-rest/Utilities/AzureAppServiceUtility');
+jest.mock('azure-actions-appservice-rest/Utilities/ContainerDeploymentUtility');
+jest.mock('azure-actions-appservice-rest/Utilities/KuduServiceUtility');
+jest.mock('azure-actions-appservice-rest/Utilities/AnnotationUtility');
+
+describe('main.ts tests', () => {
+
+    afterEach(() => {
+        jest.restoreAllMocks();
+    })
+        
+    it("gets inputs and executes function container action", async () => {
```

Comment:
Add more tests. Consider failure scenarios and different combination of inputs.

Reply:
We are setting up framework for now as discussed with Chaitanya. Will add more tests later